File: 준영/프로그래머스/LV2/N개의 최소공배수.py
def gcd(a, b):
    while b > 0:
        a, b = b, a % b
    return a

def lcm(a, b):
    return a * b / gcd(a, b)

def solution(arr):
    temp = arr[:]

    while len(temp) > 1:
        lcm_arr = []
        x = temp[0]

        for i in temp[1:]:
            lcm_arr.append(lcm(x, i))
        
        temp = lcm_arr

    answer = temp.pop()

    return answer

arr = [2,6,8,14]
solution(arr)

File: 준영/프로그래머스/LV2/N개의 최소공배수.js
function gcd(a, b) {
  while (b > 0) {
    var temp = a % b
    a = b
    b = temp
  }
  return a
}

function lcm(a, b) {
  return a * b / gcd(a, b)
}

function solution(arr) {
  var answer = 0
  
  var temp = arr.slice()
  
  while (temp.length > 1) {
    var lcm_arr = Array(0)
    var x = temp[0]

    for (let i = 1; i < temp.length; i++) {
      lcm_arr.push(lcm(x, temp[i]))
    }
    temp = lcm_arr.slice()
  }

  answer = temp[0]
  return answer
}